module/Finna/src/Finna/Service/RemsService.php

```diff
@@ -0,0 +1,826 @@
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+namespace Finna\Service;
+
+use Laminas\Config\Config;
+use Laminas\Session\Container;
+use VuFind\Auth\Manager;
+
+/**
+ * Resource Entitlement Management System (REMS) service
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+class RemsService implements
+    \VuFindHttp\HttpServiceAwareInterface,
+    \Laminas\Log\LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    // REMS Application statuses
+    const STATUS_APPROVED = 'approved';
+    const STATUS_NOT_SUBMITTED = 'not-submitted';
+    const STATUS_CLOSED = 'closed';
+    const STATUS_DRAFT = 'draft';
+    const STATUS_REVOKED = 'revoked';
+    const STATUS_REJECTED = 'rejected';
+    const STATUS_EXPIRED = 'expired';
+
+    // Session keys
+    const SESSION_IS_REMS_REGISTERED = 'is-rems-user';
+    const SESSION_ACCESS_STATUS = 'access-status';
+    const SESSION_BLACKLISTED = 'blacklisted';
+    const SESSION_USAGE_PURPOSE = 'usage-purpose';
+
+    // REMS API user types
+    const TYPE_ADMIN = 0;
+    const TYPE_APPROVER = 1;
+    const TYPE_USER = 2;
+
+    /**
+     * Configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Session container
+     *
+     * @var Container
+     */
+    protected $session;
+
+    /**
+     * Auth manager
+     *
+     * @var Manager
+     */
+    protected $auth;
+
+    /**
+     * Is the user authenticated?
+     *
+     * @var bool
+     */
+    protected $authenticated;
+
+    /**
+     * National identification number of current user (encrypted).
+     *
+     * @var string
+     */
+    protected $userIdentificationNumber = null;
+
+    /**
+     * Constructor.
+     *
+     * @param Config         $config        Configuration
+     * @param SessionManager $session       Session container
+     * @param String|null    $userId        National identification number of
```

Comment:
$userId on hämäävästi samankaltainen kuin käyttäjän kirjautumistunnus, ehkä $identityNumber?

Reply:
Jep, selkeämpi näin.